[docs/folder-structure] Changed Docs Structure / MDM Fixes

hosy opened the pull request:
<!--
Thanks for submitting a change to ownCloud!

To make it possible for us to get your change reviewed and merged please fill out below information carefully.

Please note that any kind of change first has to be submitted to the master branch which holds the next major version of ownCloud.
-->

## Description
- added missing link in MDM description
- moved ios_mdm_tables.adoc output into new folder structure
- removed no longer needed docs folder

## Related Issue
#1070 #1082 

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] Added an issue with details about all relevant changes in the [**iOS documentation repository**](https://github.com/owncloud/docs-client-ios-app/issues).
- [ ] I have read the [**CONTRIBUTING**](https://github.com/owncloud/ios-app/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Added changelog files for the fixed issues in folder changelog/unreleased

mmattel commented:
Great, many thanks!

michaelstingl commented on .github/workflows/configuration-documentation.yml:
```diff
@@ -32,4 +32,4 @@ jobs:
         with:
           commit_message: Configuration documentation updated
-          file_pattern: doc/CONFIGURATION.json docs/modules/ROOT/pages/ios_mdm_tables.adoc
+          file_pattern: doc/CONFIGURATION.json doc/antora/pages/ios_mdm_tables.adoc
```
Please generate on 2 places:

- doc/CONFIGURATION.adoc
- doc/antora/pages/mdm_tables.adoc (wie MMattel will…)

mmattel replied:
A copy (double generation) of `mdm_tables.adoc` named `doc/CONFIGURATION.adoc` does not make any sense especially because you want to combine several markdown files into `doc/CONFIGURATION.adoc` for ios-app internal documentation purposes, see your comment: https://github.com/owncloud/ios-app/pull/1094#issuecomment-1020176889. Therefore I guess this was just a mistake 😄

michaelstingl replied:
Nothing gets combined. Other `*.md`,s will just be deleted. I need the MDM content accessible in `doc/CONFIGURATION.adoc`. If you want to have the content in `doc/antora/pages/`, @hosy will add a copy for your purpose there. up2u…

michaelstingl commented:
2 tiny additions:

- Move screenshots to doc/images
- enterprise/gomplate/Branding.plist.tmpl ==> move to tools folder

michaelstingl commented:
… and can we get rid of these?

- `APP_BUILD_FLAGS.md`
- `BUILD_CUSTOMIZATION.md`
- `CONFIGURATION.md`

(and unite them all in the mighty `doc/CONFIGURATION.adoc`)

mmattel commented:
@hosy Just found by chance that we need an additional text change to have the correct doc link.

**ios-sdk/doc/CONFIGURATION.md:18**
```
The file located at `doc/CONFIGURATION.json` provides a machine-readable list of available settings.
For the `ios-sdk` repository, this includes only settings implemented in the SDK. For the full list of
settings available for configuration, please see
[`docs/modules/ROOT/pages/ios_mdm_tables.adoc`]
(https://github.com/owncloud/ios-app/blob/master/docs/modules/ROOT/pages/ios_mdm_tables.adoc)
in the `ios-app` repository.
```

hosy commented:
@mmattel @michaelstingl I did further changes in this branch, with the following changes:

- moved fastlane screenshots into doc folder
- moved gomplate branding template to tools
- deleted no longer needed documentation files (values are available in configuration.adoc)
- renamed files
- deleted antora folder
- added readme file in doc folder

Please have a look again. Thank you!

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/owncloud/ios-app?pullRequest=1094) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/owncloud/ios-app?pullRequest=1094) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: hosy<br/>:x: Matthias Hühne<br/><hr/>**Matthias Hühne** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/owncloud/ios-app?pullRequest=1094) it.</sub>

hosy commented:
@michaelstingl merge conflict is solved and you can add your review
@mmattel please add your review too.

michaelstingl commented:
> @michaelstingl merge conflict is solved and you can add your review

This is un-reviewable. Please merge, then we'll iterate from there…

mmattel commented:
> This is un-reviewable. Please merge, then we'll iterate from there…

To proceed, pls approve (2a necessary), I did already.

michaelstingl reviewed (approved):
Let's try this… 🤞
